index_copy.js

```diff
@@ -0,0 +1,104 @@
+const battleship = () => {
+
+    // create objects with the given parametrs
+    let playerOne = {
+      name: "",
+      shipCount: 0,
+      gameBoard: []
+    }
+    let playerTwo = {
+      name: "",
+      shipCount: 0,
+      gameBoard: []
+    };
+  
+    // asking for the players' name
+    playerOne.name = prompt("Player 1 what is your name?");
+    playerTwo.name = prompt("Player 2 what is your name?");
+  
+    // creating game boards
+    let boardColumn = 4;
+    let shipCountFinal = 4;
+  
+    // add game board with 0
+    for (let i = 0; i < boardColumn; i++) {
+      let boardRow = []
+      for (let j = 0; j < boardColumn; j++) {
+        boardRow.push(0)
+      }
+  
+  
+  // add rows to the end of an empty list
+      playerOne.gameBoard.push(boardRow);
+      playerTwo.gameBoard.push(boardRow);
```

Comment:
- the variable `boardRow` here is like an arrow pointing to a place in memory in the computer that holds the actual value of what's in this array
- because `boardRow` is a reference to a specific array value in memory, you are pushing into the array for playerOne and playerTwo the same reference 
- if (when) you edit one of them later, you are actually editing both of them!
this will not happen with strings and numbers because they are `primitive` data types.
this article helps explain.. but you will surely cover more of this reference vs value issue/quirk to javascript later in the course! 
https://www.dyn-web.com/javascript/arrays/value-vs-reference.php#:~:text=JavaScript%20Arrays%3A%20Value%20vs%20Reference&text=That%20means%20that%20if%20you,the%20value%20of%20the%20array.